### DISCUSSION TITLE
Your Feedback? Considering to move from AC-based charging to MPPT-based charging

### DISCUSSION THREAD

**Discussion by swingstate** [hoylabs/OpenDTU-OnBattery#249] [General]
I am considering to reconfigure my setup and ask for advice/opinions. 
My current setup is basically this: 

![Solar Setup AC charger](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/5798bd0c-99a0-4ab5-8a19-748f86d39799)

In parallel to consuming the generated solar energy I am using 3 AC/DC chargers to charge the 25,6V Battery bank, depending on how much energy excess is available. The charging is managed by Home assistant, based on what the 3em is reporting.

The main reason for this setup was/is that I cannot house the charging stack & batteries on my Terrasse. 

It works quite well, however there are 2 main disadvantages: 

1. There is still energy being wasted since I have to have approx. 125W of energy excess to switch on the smallest AC charger or to switch between the chargers. For example 100W of excess just goes to the grid, since the chargers are fixed to 4/8/16A / Every increase of charging needs an additional 125W. 

2. Home assistant has a delay to increase/decrease the charging. This can create situation where the chargers could switch from e.g. 4A to 8A, which usually happens however with a delay. Moreover, the Home assistant delay can create situations where a charger stays on, while the produced solar energy drops - consequently the charger uses energy from the grid to charge. On a bad day this can cause up to 0,6 kWh of grid consumption after having charged up to 2 kWh into the Batteries. In average the "unwanted grid consumption" stays between 0 - 0,08 kWh. 

Hence, I am considering to move my Batteries outside into an isolated & heated box (DIY), and to install a small cabinet to house the MPPTs, Inverter, breaker etc. 

I think this would look like this: 

![Solar Setup MPPT charged](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/0834b471-c477-4904-87cc-7b8cbdc0aebc)

Questions: 

1. Does this setup make sense? Any major flaws?
2. Would there be any benefit to use the VE.Direct connection from one (later 2) MPPTs to the onBattery ESP? Is it just to read the SOC?
3. What role does on-battery play exactly in this setup? I would still be using the Power Limiter function. Does on-battery also manage anything what the MPPTs do? 
4. At the moment I am using Home assistant to shut down the AC Side of the inverter when the SOC is at 20%. The DC/Battery section in on-battery does the same but reads the Voltage from the Inverter to switch it on/off, correct?

**Comment by spcqike** [comment]
i guess i would first try with one MPPT, SSO and SSW facing strings in parallel. if you can put each String in series, depends on your modules.
what kind of inrush limiter do you use? as far as i read, you only need this, if you unplug your battery. as far as your battery is always connected to your inverter, you probably won't need them. but, if you allready have them they won't harm. (afaik the main problem is, that the capacitors draw a lot of current, when connected. and the PCB/whatever aren't designed to handle such high currents, as a PV module cant deliver it anyway and won't do so in the morning)

keep in mind, that the HM-1500 can't produce 1.500W with a 25.6V battery, as its inputs are limited to ~11.5A each. also, this could eventually trip your 40A DC breaker as you could draw ~46A.

**Reply by swingstate** [reply]
I am using JaSolar panels with Voc of 41,78V and Isc of 11,53A. I am considering 2 MPPTs since the performance of the panels varies during the day, hence my assumption this would affect the overall performance. 

My panels perform like this, and the SSO facing panels have a sharp drop in performance in the afternoon due to shading. 

<img width="1217" alt="Bildschirmfoto 2023-06-01 um 21 41 28" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/d0d81678-8da6-4435-9c99-286c774607f5">

I am aware that the 1500 needs 48+ for 1475W. The Inrush Limiter I am using is the Bicker BE-DCEB10A, which only allows 30V max. and somehow seems to cut the power already when I try to pull more than 9A through it. So I am debating if I should remove this safety element, and go for 51,2V with my batteries - which are always connected to the Inverter. The low SOC disconnect happens on the AC side in my setup - works well and I consider this saver than playing on/off on the DC side. 

The draw of the inverter is really minimal. Should check that tomorrow and share here. :)

**Reply by spcqike** [reply]
What are the Panel voltages like? 

I haven’t tested this myself, as I only have two panels and a HM-600 with two inputs and MPPTs, but I read that connecting east and west facing strings in parallel at one input shouldn’t be much of a problem. The voltage rises very quickly, there is no current if the panels are shaded. But as in parallel you need the same voltage, the currents will add up. 

In Theory one 100/30 should be enough. 

2 parallel strings with 2 panels in series each, would result in ~82V/11A max per string. Even if string one only produces 50W, it should have said 82V but only 0.6A. 

As I said, I have no own experience with this. Just theory and reading. (My experience shows that my single west facing Paneel also has its 40V with little current in the early morning, but still it’s connected to its own MPPT)

**Reply by swingstate** [reply]
A photo from the type label 

<img width="494" alt="Bildschirmfoto 2023-06-01 um 22 16 31" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/178295df-235c-4051-91ec-92e1916069ad">

4 panels in series-parallel would allow to use 1 MPPT 150/35 as I understand it - when I also put the 2 batteries in series for 48V.

**Reply by spcqike** [reply]
Well yes, you may be right. I just realized that the 100/30 can’t handle 48V systems. The 100/20 can do, but it only handles around 1160W, which could limit you a little bit. As your picture shows your modules can produce up to ~350W each, which could sum up to something above 1300W in your peak time I guess.

**Reply by swingstate** [reply]
nope, they do 360-380W on good days, and up to 420 on very good cold days. I had the inverter a few times hovering at 1500W in April during cold, sunny mornings.

**Reply by swingstate** [reply]
Considering the feedback above I am arriving at this config.  
The connections from Bus bar to Inverter are simplified to keep the schematic clean.

![Solar Setup AC charger-New Setup - 1 MPPT drawio](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/f343162e-6c0d-4eb5-82a5-080c4dfc7a5e)

**Reply by spcqike** [reply]
Im not sure that your panel configuration is right. It looks like you build two strings, each having a panel facing sso and ssw. That wouldn’t work well because in one string, the current is limited to the weakest panel. 

You want one string facing sso and one string facing ssw. These two strings should be in parallel 

and I’m still not sure if your 40A DC breaker is enough to connect an HM-1500 to the battery. As, in theory, the HM-1500 can use 11.5A per input.

**Reply by swingstate** [reply]
Thanks a lot, you feedback helps a lot. 

I made couple of changes, especially with the battery system which would be a 48V setup vs. 24V.
The panels are 2 parallel strings with 2 panels in series each. I also adapted the MPPT & DC breaker accordingly. 

Even with just 35A the charging performance is higher than what I can achieve atm.

![Solar Setup AC charger-New Setup - 1 MPPT drawio](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/551f548f-0e92-4243-a211-d4225623a4e5)

**Reply by MalteSchm** [reply]
I use a single 150/35  on 4 Panels 
Your Setup makes sense

**Reply by swingstate** [reply]
How do you connect / disconnect your batteries from the inverter if needed, without risking to kill the inverter?

**Reply by swingstate** [reply]
Or do you never intend to make such disconnect @MalteSchm?

**Reply by MalteSchm** [reply]
I use a pylontech battery and just switch it on / off if needed.
Not sure about the current in this case. But it doesn't happen that often

**Comment by swingstate** [comment]
Thanks Malte!

Question regarding MPPT Performance. Sunday was I think the 2nd days this year with 0 clouds so I could track the panel output very well. 
I wonder if the shading which hits on of my SSW String panels in the afternoon would pull down the performance of the entire string and/or the MPPT dramatically? 
The impact of the shading on the SSO string is minor since this is more towards end of day.

![Screenshot 2023-06-06 at 13 33 53 copy](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/e1a18e93-b4cc-469b-b0d4-8bd32d8bacba)


At the moment both strings are connected to one MPPT input each on my HM1500.

**Reply by spcqike** [reply]
do you monitor the DC voltage, too?

i guess, the impact will small, as it is in your two strings allready.
each of your strings consists of 2 modules, which are placed in parallel. your "third" module (dark blue) gets shaded at about 18:00. but your 4th module still delivers power. 

in parallel: as long as both modules (or strings) have the same voltage, the currents will add up.

i have a HM-600, so each module has its own MPPT
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/029390ab-4347-4db2-b2ce-da9553de1c14)
as you can see, the voltages are more or less the same over the day, but the generated power is very different. (because of very different orientation and sun light)

see also here:
https://www.ikz.de/detail/news/detail/deutliche-kostenreduzierung-bei-minimalen-ertragsverlusten-ostwest-ausgerichtete-pv-anlagen-mit-n/

> Keine Nachteile
> Die Untersuchungen an den beiden PV-Anlagen haben gezeigt, dass bei einer Ost/West ausgerichteten PV-Anlage mit einem gemeinsamen Wechselrichter für den Ost- und den West-Generator Mismatching-Verluste auftreten. Wie erwartet sind diese Verluste jedoch sehr gering und werden teilweise dadurch kompensiert, dass der gemeinsame Wechselrichter meistens in einem höheren Wirkungsgradbereich arbeitet. Den minimalen Ertragsverlusten stehen deutliche Kostenreduzierungen in anderen Bereichen gegenüber: Zum einen kann die Anzahl der Wechselrichter verringert werden, und zum anderen kann die Nennleistung des gemeinsamen Wechselrichters – abhängig vom Neigungswinkel der installierten Solarmodule – um bis zu 35% reduziert werden. Darüber hinaus können auch noch die Installationskosten minimiert werden.

**Reply by swingstate** [reply]
Thanks for the hint. Looking at voltages - they seem very stable compared to the current which is 1:1 reflecting the power/irradation.

<img width="792" alt="Bildschirmfoto 2023-06-07 um 22 23 29" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/6456f526-08b1-42aa-a3c3-5949dd62cc7e">

So for a pair of 2 parallel strings with 2 panels in series each, one string facing SSO & one facing SSW should work. 

to be confirmed :)

**Comment by swingstate** [comment]
Sharing an updated drawing of the setup I am going to build, might be helpful for others. Any feedback as always welcome :)

![Solar Setup AC charger-New Setup0608 drawio](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/776e3844-1117-4ca8-a285-e9872aa1f6cd)

Couple elements to explain. 

The AC Relay measures the DC voltage and disconnects the Inverter on the AC side at 51,2V to protect the battery from discharging too deep. This solution is independent from my Smarthome (because Homeassistant might fail) and works very reliable so far in my current AC-charging setup. 

The Shelly Plus connected to the Inverter AC is the main AC-Switch for the Inverter. It is integrated into Homeassistant so I can do manuall/automated on/offs from my mobile. It also measures the energy which goes into the household.

Homeassistant, which I am using for many other smart home things, will primarily be used for day to day monitoring of the Battery SOC, Calculation of savings, Switching of the Inverter if needed. 

The Battery Soc is calculated by the Victron Shunt and reported to Homeassistant via ESPHome. I might remove the Shunt if the SOC reading & reporting from the MPPT through onBattery/MQTT is reliable. But for now I am planing to keep it. 

All active devices (Inverter, Power Plugs for ESPHome & onBattery) will be connected to a 3-way power distributor which is connected to a 16A Brennenstuhl FI Plug.

The Brennenstuhl FI Plug is the main fuse for the system if anything goes south and the connection into the Household.

**Reply by swingstate** [reply]
UI in Homeassistant looks like this on mobile & PC, still AC-charger based:

![Screenshot 2023-06-08 at 14 16 14](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/73672554/bb53214a-5567-44dc-87fd-7077221e9dff)

**Reply by spcqike** [reply]
> The AC Relay measures the DC voltage and disconnects the Inverter on the AC side at 25,75V to protect the battery from discharging too deep. This solution is independent from my Smarthome (because Homeassistant might fail) and works very reliable so far in my current AC-charging setup.

if you manage to discharge two 25.6V LiFePo4 batteries in serie to 25.75V, something is allready broken.
each battery should have its own internal BMS and protect itself. don't they switch off at something like 20V?

check out this discussion https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/248

the OP also uses two batteries in series and had some strange behaviour at the beginning. it looks like this was, because one BMS shut down (low SoC and voltage drop under load), which resulted in to little voltage for the inverter to work properly (as he sets lowest voltage level to 47V, which one battery can't deliver, of course) 

i guess, if you want to protect your batteries from over discharge and dont want to rely on the integrated BMS, you need to switch off at something like 42V allready (https://footprinthero.com/lifepo4-battery-voltage-charts). but i dont know, if i would do this. as the relay probably can't take the actual load and therefore voltage drop into account. under low SoC and high loads, this could result in strange on-off-on-off behaviour, like in #248 

a better solution would be your smart shunt or/and a smart battery protect, as the smart shunt could measure the voltage of two batteries
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/50098985/ee45b6f7-0b89-4743-88db-ae7305e0a731)

**Reply by swingstate** [reply]
good catch, 😄  the 25,75 is from my current setup. With the 2 in series it will be 51,2V. Post adjusted.

Basically I am stopping the discharge at 18-20%. 

Regarding Smart Shunt - Yes I will use my shunt as you shared above. The Relay PCB is just a 2nd layer of protection, if Homeassistant would fail. Basically a fallback solution. And then there is the internal BMS of every battery which would stop the discharge as a lost resort at 21,6V / 43,2V which is the equivalent to a Soc of 1%.

**Reply by swingstate** [reply]
Another layer of protection is the SOC reading from the MPPT, which to my understanding can be used in onBattery to stop the inverter.

**Reply by Manos1966** [reply]
I have an unqualified question:
Will the Smartshunt correctly identify the charge status of the batteries from the voltage **during daytime,** while VICTRON MPPT is charging the batteries? 
If I understand correctly, the VICTRON MPPT is charging with a higher voltage, and this is the voltage you will see at the Inputs of your HM-1500 thus leading you to the wrong assumption that the battery is full.

The question even is even more pertinent while trying to define the cut-off of the battery ( stopping the discharge at 18-20%.), during daytime, while the VICTRON MPPT is charging the battery.

During daytime: You have 4 x 385 =  1540W peak at best. 
Your HM-1500 will go up to 1500W thus, there may be moments it will be providing more energy than the VICTRON at that moment can provide. So, it may take the extra energy from the battery. So far, so good.
Now, let's suppose your battery is empty from last night's usage. You should stop any further discharge.
However, since the VICTRON MPPT is charging with higher voltage, the voltage at the poles of the battery is higher and above the cut-off.
Subsequently the OpenDTU will give the go ahead to the Hoymiles to produce, which could be up to 1500W. 
As mentioned earlier, the VICTRON MPPT can not deliver the full 1500W (unlikely) thus the battery will be drained and soon drops below the cut-off voltage. So, the cut off is activated, battery drainage stops, VICTRON MPPT continues charging thus voltage is again higher than cur-off  --- and the cycle starts again!

At least this is how it was before yesterday's Firmware. I have to test if the new parameters can help avoid this vicious loop.
I am also waiting for a Pylontech battery in the hope that using the SOC from the CANbus will help avoiding this loop.

**Reply by spcqike** [reply]
@Manos1966 that’s what SoC is for. 

State of charge is calculated, based on how much energy goes in or out. 

As far as I understood you need to fully charge your battery, tell your smartshunt „it’s full. 100% SoC, and it’s total capacity is 100Ah“. Then it calculates the SoC based on current flow / voltage -> power / energy. 

Let’s assume you draw 10A for 1 hour. That’s 10Ah. So it should report 90% SoC as remaining. 

If you then charge with 5A for 30 minute that’s 2.5Ah, you should see 92.5% SoC again. 

That’s also how Smartphones and thing calculate the little battery icon. As voltage alone as indicator is not enough. 

If your real capacity goes down (which happens after a while also when charging from 50%-100%), your phone can die at like 10% SoC. After a full charge the monitor chip knows the real capacity, again. And calculate the the SoC precise again.

**Comment by Manos1966** [comment]
> The main reason for this setup was/is that I cannot house the charging stack & batteries on my Terrasse.

I have the same opinion about this and I think there is a solution thanks to this amazing OpenDTUonBattery:

> Huawei R4850G2 power supply unit that can act as AC charger. Supports status shown on the web interface and options to set voltage and current limits on the web interface and via MQTT. Connection is done using CAN bus (needs to be separate from Pylontech CAN bus) via SN65HVD230 interface.

(have a Google check on the Huawei R4850G2) 👍 

 Also, regarding the performance of the HM-1500, here are the results of my tests: #244

**Reply by swingstate** [reply]
I am sticking to 2x 25,6 100Ah in parallel as 1x 25,6V 200Ah Battery, since I don't want to balance the batteries every 6 months which would mean to pull everything more or less apart. For now approx 1140W max. (26V x 11A = 286W x 4 channel) are ok for my case. I am more interested to cover by base load and when my kids use their X-Box (1 X-Box One X / Series X is 150W each). 

I also couldn't find a battery balancer for 48/51V that works with 2x 25,6V Batteries.

**Reply by Manos1966** [reply]
I just add this from the other discussion we had #529 just to keep things together...
Apparently there is one/two models:
HB02-2S24V or HA02-2S24V
![Battery Balancer 2x24V](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/7fe6a6f4-8954-4548-8348-288d47568204)

**Reply by swingstate** [reply]
The HA02 seems to be a pretty old model, suited for 12V Batteries only.

**Reply by Manos1966** [reply]
The 2S24V is the defining factor = 2 Strings, 24V per string
